src/opensearch_sdk_py/actions/response_handlers.py

```diff
@@ -23,6 +23,8 @@ def register(self, request_id: int, handler: ResponseHandler) -> None:
         self[request_id] = handler
 
     def handle(self, response: OutboundMessageResponse, input: StreamInput = None) -> Optional[bytes]:
-        handler = self[response.request_id]
-        del self[response.request_id]
-        return handler.handle(response, input) if handler else None
+        if response.request_id in self:
```

Comment:
This is a double lookup, does `handler = self[response.request_id]` or `handler = get(response.request_id, None)` work? Then you can `if` that.

Reply:
`handler = self[response.request_id]` will throw a `KeyError` exception, and I generally try to avoid using exceptions as logic in code.

Not familiar with `get()` but if it's equivalent to Java 8+ `Map.getOrDefault()` then that would work as far as returning a `None` handler, but the deletion from the map would be irrelevant and I'd have to "None-check" the handler before calling the handle method, so I don't see any value here.

Reply:
What module is `get()` in?

Reply:
I believe `Dict` in python has it, https://stackoverflow.com/questions/46024562/how-do-i-avoid-keyerror-when-working-with-dictionaries, but I am compiling this in my head, didn't try.

Reply:
Ah, I see it now.  Yes, it would have worked:
```python
def handle(self, response: OutboundMessageResponse, input: StreamInput = None) -> Optional[bytes]:
    handler = self.get(response.request_id)
    if handler is not None:
        del self[response.request_id]
        return handler.handle(response, input) if handler else None
    return None
```
But I think since we're removing the key from the dict as part of the logic it's more readable this way.

Although using `self.pop(response.request_id, None)` would have done the removal and is probably the best choice... but I already merged it. :)